src/Storage/Storage.Management/StorageAccount/SetAzureStorageAccount.cs

```diff
@@ -680,77 +697,93 @@ public override void ExecuteCmdlet()
+                            PublishMicrosoftEndpoints = this.publishMicrosoftEndpoint,
+
+                        };
                     }
                     if (allowSharedKeyAccess != null)
                     {
-                        updateParameters.AllowSharedKeyAccess = allowSharedKeyAccess;
+                        storageAccountPatch.AllowSharedKeyAccess = this.allowSharedKeyAccess;
                     }
                     if (sasExpirationPeriod != null)
                     {
-                        updateParameters.SasPolicy = new SasPolicy(sasExpirationPeriod.Value.ToString(@"d\.hh\:mm\:ss"));
+                        storageAccountPatch.SasPolicy = new Track2Models.SasPolicy(
+                            this.sasExpirationPeriod.Value.ToString(@"d\.hh\:mm\:ss"), null);
                     }
                     if (keyExpirationPeriodInDay != null)
                     {
-                        updateParameters.KeyPolicy = new KeyPolicy(keyExpirationPeriodInDay.Value);
+
+                        storageAccountPatch.KeyExpirationPeriodInDays = keyExpirationPeriodInDay.Value;
+
+
                     }
                     if (allowCrossTenantReplication != null)
                     {
-                        updateParameters.AllowCrossTenantReplication = allowCrossTenantReplication;
+                        storageAccountPatch.AllowCrossTenantReplication = this.allowCrossTenantReplication;
                     }
                     if (this.PublicNetworkAccess != null)
                     {
-                        updateParameters.PublicNetworkAccess = this.PublicNetworkAccess;
+                        storageAccountPatch.PublicNetworkAccess = this.PublicNetworkAccess;
                     }
-                    if(this.immutabilityPeriod !=null ||  this.ImmutabilityPolicyState != null)
+                    if (this.immutabilityPeriod != null || this.ImmutabilityPolicyState != null)
                     {
-                        updateParameters.ImmutableStorageWithVersioning = new ImmutableStorageAccount();
-                        updateParameters.ImmutableStorageWithVersioning.ImmutabilityPolicy = new AccountImmutabilityPolicyProperties();
-                        updateParameters.ImmutableStorageWithVersioning.ImmutabilityPolicy.ImmutabilityPeriodSinceCreationInDays = this.immutabilityPeriod;
-                        updateParameters.ImmutableStorageWithVersioning.ImmutabilityPolicy.State = this.ImmutabilityPolicyState;
+                        storageAccountPatch.ImmutableStorageWithVersioning = new Track2Models.ImmutableStorageAccount();
+                        storageAccountPatch.ImmutableStorageWithVersioning.ImmutabilityPolicy = new Track2Models.AccountImmutabilityPolicyProperties
+                        {
+                            ImmutabilityPeriodSinceCreationInDays = this.immutabilityPeriod,
+                            State = this.ImmutabilityPolicyState,
+                        };
```

Comment:
New Track2Models.AccountImmutabilityPolicyProperties will fail with following exception, when not set "ImmutabilityPolicyState" in the cmdlet parameter, but set "ImmutabilityPeriodSinceCreationInDays" . 
We should allow user only set  "ImmutabilityPeriodSinceCreationInDays" and keep "ImmutabilityPolicyState" not changed on server. This should be an SDK issue. please record it.

> System.ArgumentNullException
>   HResult=0x80004003
>   Message=Value cannot be null.
> Parameter name: value
>   Source=Azure.ResourceManager.Storage
>   StackTrace:
>    at Azure.ResourceManager.Storage.Models.AccountImmutabilityPolicyState..ctor(String value)
>    at Azure.ResourceManager.Storage.Models.AccountImmutabilityPolicyState.op_Implicit(String value)
>    at Microsoft.Azure.Commands.Management.Storage.SetAzureStorageAccountCommand.ExecuteCmdlet() in D:\code\PSH_Dev\src\Storage\Storage.Management\StorageAccount\SetAzureStorageAccount.cs:line 776
>    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
>

Reply:
Fixed and recorded.